src/java/org/apache/cassandra/utils/caching/TinyThreadLocalPool.java

```diff
@@ -46,10 +48,23 @@ public void offer(V value)
         }
         private void offerSafe(V value)
         {
+            if (DEBUG)
```

Comment:
if you have 3 rows in a update, we will offer the same reference 3 times!  This check now detects that and fails; but you must be in "debug mode" to hit that... I updated `CQLTester` and jvm-dtest to be in debug mode so we should see these cases in tests 🤞